yamls/extension/test-extension-deployment.yaml

```
@@ -0,0 +1,138 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: runtime-extension-test
+  name: test-extension-system
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: runtime-extension-test
+  name: test-extension-manager
+  namespace: test-extension-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: runtime-extension-test
+  name: test-extension-manager-role
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  verbs:
+  - get
+  - list
+  - watch
+  - patch
+  - update
+  - create
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: runtime-extension-test
+  name: test-extension-manager-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: test-extension-manager-role
+subjects:
+- kind: ServiceAccount
+  name: test-extension-manager
+  namespace: test-extension-system
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: runtime-extension-test
+  name: test-extension-webhook-service
+  namespace: test-extension-system
+spec:
+  ports:
+  - port: 443
+    targetPort: webhook-server
+  selector:
+    app: test-extension-manager
+    cluster.x-k8s.io/provider: runtime-extension-test
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: runtime-extension-test
+  name: test-extension-manager
+  namespace: test-extension-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: test-extension-manager
+      cluster.x-k8s.io/provider: runtime-extension-test
+  template:
+    metadata:
+      labels:
+        app: test-extension-manager
+        cluster.x-k8s.io/provider: runtime-extension-test
+    spec:
+      containers:
+      - command:
+        - /manager
+        image: gcr.io/kakaraparthy-devel/test-extension:v1.0.0
+        imagePullPolicy: Always
```

```suggestion
        imagePullPolicy: IfNotPresent
```
(necessary for the offline case, doesn't hurt even if run without prepull & with internet)